File: 2022/UICPC DIV1/Contests/UICPC Round #04 (Div 1)/solutions/Problem K - Tractor/tractor.py
import math
t = int(input())
for i in range(t):
    n, m = map(int, input().split())
    if n > m:
        n, m = m, n
    lg = math.log2(m+1)
    x = int(lg)
    res = max(0, n + m + 2 - 2 ** (x+1))
    for j in range(x+1):
        res += min(n+1, 2 ** j)
    print(res)

File: 2022/UICPC DIV1/Contests/UICPC Round #04 (Div 1)/solutions/Problem K - Tractor/tractor.cpp
#include <bits/stdc++.h>
using namespace std;
#define EPS 1e-9
long long int n, m;

int main()
{
    int t;
    cin >> t;
    while(t--){
        cin >> n >> m;
        if(n > m){
            swap(n, m);
        }
        double lg = log2(m+1);
        int x = (int)lg;
        long long int res = max(n + m + 2 - (long long int)pow(2, x+1), 0LL);
        for (int i = 0; i <= x; ++i) {
            res += min(n+1, (long long int)pow(2, i));
        }
        cout << res << endl;
    }
    return 0;
}

